### PR TITLE
[dataprotection] Support for Azure Kubernetes Service + Migrate parts of the `backup-instance` subcommand to AAZ-dev-tools

### PR DESCRIPTION
**Pull request by zubairabid**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

`az dataprotection backup-instance`

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
dataprotection

**Comment by zubairabid** [comment]
Hi @evelyn-ys @calvinhzy can you please help identify why the `test_dataprotection_oss` and the `test_dataprotection_aks` tests are failing with `... has no attribute 'kwargs'`? These are both tests that are passing on my local machine, after using the latest cli and extensions dev/main branches.

File: src/dataprotection/azext_dataprotection/tests/latest/test_dataprotection_aks_workload.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        raise e
+    finally:
+        cleanup(test)
+
+# Test class for Scenario
+class DataprotectionScenarioTest(ScenarioTest):
+    def __init__(self, *args, **kwargs):
+        super(DataprotectionScenarioTest, self).__init__(*args, **kwargs)
+
+    def test_dataprotection_aks(self):
```
Comment (kairu-ms) [inline src/dataprotection/azext_dataprotection/tests/latest/test_dataprotection_aks_workload.py:230]: You should add `@AllowLargeResponse()` before this function

File: src/dataprotection/azext_dataprotection/tests/latest/test_dataprotection_aks_workload.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    setup(test)
+    try:
+        setup_vault_and_policy(test)
+        configure_backup(test)
+        trigger_backup(test)
+        trigger_restore_original_location(test)
+    except Exception as e:
+        raise e
+    finally:
+        cleanup(test)
```
Comment (kairu-ms) [inline src/dataprotection/azext_dataprotection/tests/latest/test_dataprotection_aks_workload.py:223]: Why not move them under the test_dataprotection_aks?

Comment (zubairabid) [reply src/dataprotection/azext_dataprotection/tests/latest/test_dataprotection_aks_workload.py:223]: In this file I am following the pattern that we already have in `test_dataprotection_oss_workload.py`

File: src/dataprotection/azext_dataprotection/tests/latest/test_dataprotection_aks_workload.py (hunk 1 above)
Comment (zubairabid) [reply src/dataprotection/azext_dataprotection/tests/latest/test_dataprotection_aks_workload.py:230]: Updated.

File: src/dataprotection/azext_dataprotection/generated/commands.py [hunk 3]
```diff
@@ -80,20 +80,20 @@ def load_command_table(self, _):
         'dataprotection backup-instance', dataprotection_backup_instance, client_factory=cf_backup_instance
     ) as g:
         g.custom_command('list', 'dataprotection_backup_instance_list')
-        g.custom_show_command('show', 'dataprotection_backup_instance_show')
-        g.custom_command('create', 'dataprotection_backup_instance_create', supports_no_wait=True)
-        g.custom_command('delete', 'dataprotection_backup_instance_delete', supports_no_wait=True, confirmation=True)
+        # g.custom_show_command('show', 'dataprotection_backup_instance_show')
+        # g.custom_command('create', 'dataprotection_backup_instance_create', supports_no_wait=True)
+        # g.custom_command('delete', 'dataprotection_backup_instance_delete', supports_no_wait=True, confirmation=True)
```
Comment (kairu-ms) [inline src/dataprotection/azext_dataprotection/generated/commands.py:85]: Can you clean the code

Comment (zubairabid) [reply src/dataprotection/azext_dataprotection/generated/commands.py:85]: As the `backup-instance` subgroup has not fully migrated yet, I have left the migrated commands in as reference. The idea is that when the full migration is completed, this code will be cleaned up properly. Is that alright, or should I just remove these lines?

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**